hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/clientrm/FederationClientInterceptor.java

```diff
@@ -949,7 +949,7 @@ public ReservationSubmissionResponse submitReservation(
         // Second, determine whether the current ReservationId has a corresponding subCluster.
         // If it does not exist, add it. If it exists, update it.
         Boolean exists = existsReservationHomeSubCluster(reservationId);
-        if (!exists) {
+        if (!exists || i == 0) {
```

Comment:
What is this for?

Reply:
Reservation may be repeatedly submitted (the same reservationId is submitted repeatedly). At this time, we should try to use the first reservation result. If the first reservation encounters some abnormalities, find another RM for reservation.